bokehjs/src/coffee/renderer/glyph/image_url.coffee

```diff
@@ -4,35 +4,41 @@ Glyph = require "./glyph"
 class ImageURLView extends Glyph.View
 
   _index_data: () ->
-    @_xy_index()
 
   _set_data: () ->
-    @image = (null for img in @url)
-    @need_load = (true for img in @url)
-    @loaded = (false for img in @url)
-    @_xy_index()
+    if not @image? or @image.length != @url.length
+      @image = (null for img in @url)
+
+    for i in [0...@url.length]
+      img = new Image()
+      img.onload = do (img, i) =>
+        return () =>
+          @image[i] = img
+          @renderer.request_render()
+      img.src = @url[i]
```

Comment:
Came in with the ugly merge, not sure why it's showing up since it is what is in master currently AFAIK.